src/P/Either.hs

```diff
@@ -24,3 +34,11 @@ ecase e l = flip (either l) e
 
 flipEither :: Either a b -> Either b a
 flipEither = either Right Left
+
+lefts :: (Foldable f) => f (Either a b) -> [a]
+lefts =
+  Either.lefts . toList
+
+rights :: (Foldable f) => f (Either a b) -> [b]
```

Comment:
Consider adding `SPECIALIZE` pragmas for these

Reply:
Could you give the full `SPECIALIZE` line, please? I've never used it before.
(Given I'd need to give it a more specialised type, I'm not sure what to put.)

Reply:
`{-# SPECIALISE lefts :: [Either a b] -> [a] #-}`

Reply:
AH, right. And so it'll still work for other types, but not be optimised.

Reply:
(Added.)